src/poetry/utils/env/script_strings.py

```diff
@@ -54,12 +28,7 @@ def interpreter_version():
 
 
 def _version_nodot(version):
-    if any(v >= 10 for v in version):
-        sep = "_"
-    else:
-        sep = ""
-
-    return sep.join(map(str, version))
+    return "".join(map(str, version))
```

Comment:
Had this always been wrong and we just did not use it?

Reply:
I was producing a wrong nodot version. I spent some time debugging why there was a difference between old and new implementation and that was the culprit. This is the implementation that `packaging` is using for that function.